src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,51 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object
+     *
+     * @param string $key      The parameter key
+     * @param string $format   The expected date format
+     * @param mixed  $default  The default value if the parameter key does not exist
```

Comment:
The default should be a DateTime or null here. Passing an integer will break the contract of the return value